blob/fileblob/fileblob.go

```diff
@@ -74,15 +74,26 @@ func init() {
 }
 
 // Options sets options for constructing a *blob.Bucket backed by fileblob.
-type Options struct{}
+type Options struct {
+	URLSigner URLSigner
+}
 
 type bucket struct {
-	dir string
+	dir       string
+	urlSigner URLSigner
+}
+
+func (b *bucket) path(key string) (string, error) {
```

Comment:
Nit: This refactor (pulling out `path`) looks good, but it's unrelated to the rest of this PR, can you pull it out into a separate PR?